### PR TITLE
chore: Get rid of .localized and replace it with L10n.Localizable struct - P2

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

## Summary

This PR addresses the task of standardizing localization practices across the codebase. We have replaced all instances of the .localized method with the newly generated L10n.Localizable struct, which provides type-safe access to localized strings.

## Changes

Updated localization calls from .localized to L10n.Localizable to ensure consistency and safety.
Enhanced the maintainability of localization by utilizing SwiftGen to generate L10n.Localizable, reducing the risk of typos and simplifying the process of updating localized strings.

## Affected files

- `ConversationInputBarViewController+FilesPopover.swift`
- `ConversationInputBarViewController+SendButton.swift`
- `ConversationInputBarViewController.swift`
- `EphemeralKeyboardViewController.swift`
- `ConversationListViewController.swift`
- `ConversationListTopBarViewController.swift`
- `ConversationListViewModel.swift`
- `ConversationListOnboardingHint.swift`
- `ZMConversation+Status.swift`
- `PasscodeSetupViewController.swift`
- `AppLockChangeWarningViewController.swift`
- `WipeCompletionViewController.swift`
- `WipeDatabasePresenter.swift`
- `FolderCreationNameSectionController.swift`
- `FolderPickerController.swift`
- `ConversationActionController+Block.swift`
- `ConversationActionController+ClearContent.swift`
- `ConversationActionController+DeleteGroup.swift`
- `ConversationActionController+Leave.swift`
- `ConversationActionController+Notifications.swift`
- `UIAlertController+RemoveAction.swift`
- `GroupOptionsSectionController.swift`
- `ParticipantsSectionController.swift`
- `ReceiptOptionsSectionController.swift`
- `RenameGroupSectionController.swift`
- `ServicesSectionController.swift`
- `LegalHoldAlertFactory.swift`
- `RequestPasswordController.swift`
- `GroupDetailsFooterView.swift`

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift [hunk 1]
```diff
@@ -356,7 +356,7 @@ final class SecurityAlertMatcher: ConversationStatusMatcher {
            let messageText = data.messageText {
             text = messageText
         } else {
-            text = "conversation.status.poll.default".localized
+            text = ""
```
Comment (agisilaos) [inline wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift:359]: `conversation.status.poll.default".localized` doesn't exist so for now I just removed it.

Comment (KaterinaWire) [inline wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift:359]: Why did you remove?

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift [hunk 2]
```diff
@@ -300,9 +300,9 @@ final class CallingMatcher: ConversationStatusMatcher {
     func description(with status: ConversationStatus, conversation: MatcherConversation) -> NSAttributedString? {
         if conversation.voiceChannel?.state.canJoinCall == true {
             if let callerDisplayName = conversation.voiceChannel?.initiator?.name {
-                return "conversation.status.incoming_call".localized(args: callerDisplayName) && type(of: self).regularStyle
+                return L10n.Localizable.Conversation.Status.incomingCall(callerDisplayName) && type(of: self).regularStyle
             } else {
-                return "conversation.status.incoming_call.someone".localized && type(of: self).regularStyle
+                return L10n.Localizable.Conversation.Status.IncomingCall.unknown && type(of: self).regularStyle
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift]: Shouldn't be` .someone`?

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift:359]: It didn't exist in the codebase. @KaterinaWire

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift]: Fixed it! Good catch!

File: wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/Sections/GroupOptionsSectionController.swift [hunk 3]
```diff
@@ -81,7 +81,7 @@ final class GroupOptionsSectionController: GroupDetailsSectionController {
     // MARK: - Collection View
 
     override var sectionTitle: String {
-        return "participants.section.settings".localized(uppercased: true)
+        return L10n.Localizable.Participants.Section.settings.localizedUppercase
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/Sections/GroupOptionsSectionController.swift:84]: By the way I had a talk with Astrid yesterday and she said that this "uppercase" does not work in all languages. So it would be better to let the copywriters handle it manually.
No need to change here now, but as a general information 😄

**Review by markusfassbender — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/Sections/GroupOptionsSectionController.swift (hunk 3 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/Sections/GroupOptionsSectionController.swift:84]: Thank you for the heads up. I will gather all appropriate information and I will handle it on a separate PR.

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+SendButton.swift [hunk 4]
```diff
@@ -43,8 +43,8 @@ extension ConversationInputBarViewController {
     func showAlertIfTextIsTooLong(text: String) -> Bool {
         guard text.count > SharedConstants.maximumMessageLength else { return false }
 
-        let alert = UIAlertController.alertWithOKButton(title: "conversation.input_bar.message_too_long.title".localized,
-                                                        message: "conversation.input_bar.message_too_long.message".localized(args: SharedConstants.maximumMessageLength))
+        let alert = UIAlertController.alertWithOKButton(title: L10n.Localizable.Conversation.InputBar.MessageTooLong.title,
+                                                        message: L10n.Localizable.Conversation.InputBar.MessageTooLong.message(SharedConstants.maximumMessageLength))
```
Comment (theswiftway) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+SendButton.swift]: Nitpick : Please format this to be on separate lines

Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+SendButton.swift]: Done! @theswiftway

**Review by theswiftway — APPROVED** [review]
LGTM
